<a href="https://colab.research.google.com/github/nxzK1/challenge-AluraStore/blob/desarrollo%2Fprodvendidos-promedioenvios/AluraStoreLatam.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### Importación de datos



In [2]:
import pandas as pd

url = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_1%20.csv"
url2 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_2.csv"
url3 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_3.csv"
url4 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_4.csv"

tienda = pd.read_csv(url)
tienda2 = pd.read_csv(url2)
tienda3 = pd.read_csv(url3)
tienda4 = pd.read_csv(url4)

tienda.head()

Producto Categoría del Producto    Precio  Costo de envío  \
0  Asistente virtual           Electrónicos  164300.0          6900.0   
1    Mesa de comedor                Muebles  192300.0          8400.0   
2      Juego de mesa               Juguetes  209600.0         15900.0   
3         Microondas      Electrodomésticos  757500.0         41000.0   
4   Silla de oficina                Muebles  335200.0         20200.0   

  Fecha de Compra         Vendedor Lugar de Compra  Calificación  \
0      16/01/2021      Pedro Gomez          Bogotá             4   
1      18/05/2022  Beatriz Morales        Medellín             1   
2      15/03/2021   Juan Fernandez       Cartagena             1   
3      03/05/2022   Juan Fernandez            Cali             4   
4      07/11/2020    Maria Alfonso        Medellín             5   

       Método de pago  Cantidad de cuotas       lat       lon  
0  Tarjeta de crédito                   8   4.60971 -74.08175  
1  Tarjeta de crédito                   4   6.25184 -75.56359  
2  Tarjeta de crédito                   1  10.39972 -75.51444  
3               Nequi                   1   3.43722 -76.52250  
4               Nequi                   1   6.25184 -75.56359

In [3]:
tienda2.head()

Producto  Categoría del Producto    Precio  Costo de envío  \
0  Silla de oficina                 Muebles  283600.0         15300.0   
1         Pandereta  Instrumentos musicales   76800.0          2300.0   
2         Cama king                 Muebles  443300.0         23800.0   
3         Cama king                 Muebles  906200.0         46400.0   
4  Silla de oficina                 Muebles  397800.0         19400.0   

  Fecha de Compra         Vendedor Lugar de Compra  Calificación  \
0      04/08/2021  Izabela de León        Medellín             1   
1      26/11/2020   Blanca Ramirez        Medellín             3   
2      22/10/2021  Izabela de León         Pereira             5   
3      04/05/2022    Maria Alfonso            Cali             5   
4      26/02/2022    Lucas Olivera          Bogotá             4   

       Método de pago  Cantidad de cuotas      lat       lon  
0               Nequi                   1  6.25184 -75.56359  
1  Tarjeta de crédito                   4  6.25184 -75.56359  
2  Tarjeta de crédito                  10  4.81333 -75.69611  
3  Tarjeta de crédito                  10  3.43722 -76.52250  
4  Tarjeta de crédito                   3  4.60971 -74.08175

In [4]:
tienda3.tail()

Producto Categoría del Producto    Precio  \
2354  Dashboards con Power BI                 Libros   26700.0   
2355  Dashboards con Power BI                 Libros   31800.0   
2356         Silla de oficina                Muebles  477100.0   
2357           Mesa de centro                Muebles  123100.0   
2358      Modelado predictivo                 Libros   63400.0   

      Costo de envío Fecha de Compra         Vendedor Lugar de Compra  \
2354          2100.0      21/10/2022  Ángel Rodriguez          Bogotá   
2355          2200.0      23/06/2022   Blanca Ramirez         Pereira   
2356         25600.0      19/02/2023    Camila Rivera       Cartagena   
2357          4700.0      19/09/2021   Blanca Ramirez          Bogotá   
2358          9400.0      24/05/2022  Mariana Herrera        Medellín   

      Calificación      Método de pago  Cantidad de cuotas       lat       lon  
2354             5  Tarjeta de crédito                   1   4.60971 -74.08175  
2355             5  Tarjeta de crédito                   1   4.81333 -75.69611  
2356             5  Tarjeta de crédito                   3  10.39972 -75.51444  
2357             4  Tarjeta de crédito                   8   4.60971 -74.08175  
2358             5  Tarjeta de crédito                  10   6.25184 -75.56359

In [5]:
tienda4.tail()

Producto   Categoría del Producto    Precio  Costo de envío  \
2353     Mesa de comedor                  Muebles  227700.0         12600.0   
2354    Silla de oficina                  Muebles  342000.0         18600.0   
2355        Vaso térmico     Deportes y diversión   36700.0          2400.0   
2356        Set de ollas  Artículos para el hogar  151800.0          8300.0   
2357  Cuerda para saltar     Deportes y diversión   11200.0          1300.0   

     Fecha de Compra        Vendedor Lugar de Compra  Calificación  \
2353      12/02/2020   Camila Rivera        Medellín             5   
2354      17/05/2022   Felipe Santos     Bucaramanga             5   
2355      03/05/2022   Camila Rivera          Bogotá             5   
2356      15/06/2020  Blanca Ramirez            Cali             5   
2357      17/11/2021   Felipe Santos          Bogotá             5   

          Método de pago  Cantidad de cuotas      lat       lon  
2353  Tarjeta de crédito                   1  6.25184 -75.56359  
2354  Tarjeta de crédito                   3  7.12539 -73.11980  
2355  Tarjeta de crédito                   1  4.60971 -74.08175  
2356  Tarjeta de crédito                   1  3.43722 -76.52250  
2357  Tarjeta de crédito                   1  4.60971 -74.08175

#1. Análisis de facturación



En este primer análisis, debes calcular el ingreso total de cada tienda. **Sumando los valores de la columna Precio** de cada conjunto de datos de la tienda para estimar los ingresos.

In [6]:
print(sum(tienda.Precio))
print(sum(tienda2.Precio))
print(sum(tienda3.Precio))
print(sum(tienda4.Precio))


1150880400.0
1116343500.0
1098019600.0
1038375700.0


In [53]:
def total_facturado(tienda) -> int:
  suma_total = int(sum(tienda))
  return int(suma_total)

print(f"Los ingresos estimados por la tienda 1 fueron: {total_facturado(tienda.Precio)}")
print(f"Los ingresos estimados por la tienda 2 fueron: {total_facturado(tienda2.Precio)}")
print(f"Los ingresos estimados por la tienda 3 fueron: {total_facturado(tienda3.Precio)}")
print(f"Los ingresos estimados por la tienda 4 fueron: {total_facturado(tienda4.Precio)}")


Los ingresos estimados por la tienda 1 fueron: 1150880400
Los ingresos estimados por la tienda 2 fueron: 1116343500
Los ingresos estimados por la tienda 3 fueron: 1098019600
Los ingresos estimados por la tienda 4 fueron: 1038375700


In [55]:
nombre_tiendas = ['Tienda 1', 'Tienda 2', 'Tienda 3', 'Tienda 4']
facturacion = [total_facturado(tienda.Precio), total_facturado(tienda2.Precio), total_facturado(tienda3.Precio), total_facturado(tienda4.Precio)]

print(nombre_tiendas, facturacion)

['Tienda 1', 'Tienda 2', 'Tienda 3', 'Tienda 4'] [1150880400, 1116343500, 1098019600, 1038375700]


In [56]:
facturacion_tienda = list(zip(nombre_tiendas, facturacion))
facturacion_tienda

[('Tienda 1', 1150880400),
 ('Tienda 2', 1116343500),
 ('Tienda 3', 1098019600),
 ('Tienda 4', 1038375700)]

In [57]:
print(f"La tienda que generó mas ventas fue: {max(facturacion)}")
print(f"La tienda que generó menos ventas fue: {min(facturacion)}")

La tienda que generó mas ventas fue: 1150880400
La tienda que generó menos ventas fue: 1038375700


# 2. Ventas por categoría

En este paso, vas a agrupar los datos por categoría y contar cuántas ventas hubo de cada tipo. Esto te permitirá identificar las categorías más populares por tienda.

In [11]:
print(tienda['Categoría del Producto'])

def obtener_categoria(tienda):
  categoria_producto = tienda
  return categoria_producto

tienda1_categoria = obtener_categoria(tienda['Categoría del Producto'])
tienda2_categoria = obtener_categoria(tienda2['Categoría del Producto'])
tienda3_categoria = obtener_categoria(tienda3['Categoría del Producto'])
tienda4_categoria = obtener_categoria(tienda4['Categoría del Producto'])

0            Electrónicos
1                 Muebles
2                Juguetes
3       Electrodomésticos
4                 Muebles
              ...        
2354         Electrónicos
2355         Electrónicos
2356             Juguetes
2357              Muebles
2358         Electrónicos
Name: Categoría del Producto, Length: 2359, dtype: object


In [12]:
def contador_categorias(tienda):
  contador = tienda.value_counts()
  return contador

listado_cat_t1 = [contador_categorias(tienda1_categoria)]
listado_cat_t2 = [contador_categorias(tienda2_categoria)]
listado_cat_t3 = [contador_categorias(tienda3_categoria)]
listado_cat_t4 = [contador_categorias(tienda4_categoria)]

In [13]:
listado_cat_tiendas = [listado_cat_t1, listado_cat_t2, listado_cat_t3, listado_cat_t4]
listado_cat_tiendas

[[Categoría del Producto
  Muebles                    465
  Electrónicos               448
  Juguetes                   324
  Electrodomésticos          312
  Deportes y diversión       284
  Instrumentos musicales     182
  Libros                     173
  Artículos para el hogar    171
  Name: count, dtype: int64],
 [Categoría del Producto
  Muebles                    442
  Electrónicos               422
  Juguetes                   313
  Electrodomésticos          305
  Deportes y diversión       275
  Instrumentos musicales     224
  Libros                     197
  Artículos para el hogar    181
  Name: count, dtype: int64],
 [Categoría del Producto
  Muebles                    499
  Electrónicos               451
  Juguetes                   315
  Electrodomésticos          278
  Deportes y diversión       277
  Libros                     185
  Instrumentos musicales     177
  Artículos para el hogar    177
  Name: count, dtype: int64],
 [Categoría del Producto
  Muebles         

# 3. Calificación promedio de la tienda


En este paso del challenge, el objetivo es calcular el promedio de las evaluaciones para entender qué tan satisfechos están los clientes.

In [14]:
def obtener_categoria(tienda):
  categoria_producto = tienda
  return categoria_producto

calificacion_tienda1 = obtener_categoria(tienda['Calificación'])
calificacion_tienda2 = obtener_categoria(tienda2['Calificación'])
calificacion_tienda3 = obtener_categoria(tienda3['Calificación'])
calificacion_tienda4 = obtener_categoria(tienda4['Calificación'])

In [15]:
def calcula_promedio(lista) -> float:
  resultado = sum(lista)/len(lista)
  return resultado

promedio_tienda1 = round(calcula_promedio(calificacion_tienda1),2)
promedio_tienda2 = round(calcula_promedio(calificacion_tienda2),2)
promedio_tienda3 = round(calcula_promedio(calificacion_tienda3),2)
promedio_tienda4 = round(calcula_promedio(calificacion_tienda4),2)

In [16]:
promedio_tiendas = [['Tienda: 1', promedio_tienda1], ['Tienda 2', promedio_tienda2], ['Tienda 3', promedio_tienda3], ['Tienda 4: ', promedio_tienda4]]
promedio_tiendas

[['Tienda: 1', 3.98],
 ['Tienda 2', 4.04],
 ['Tienda 3', 4.05],
 ['Tienda 4: ', 4.0]]

# 4. Productos más y menos vendidos

Como cada fila representa una unidad vendida, el secreto está en contar cuántas veces aparece cada producto en los datos de cada tienda.

In [21]:
cantidad_productos_t1 = contador_categorias(tienda.Producto)
cantidad_productos_t1

Producto
Microondas                     60
TV LED UHD 4K                  60
Armario                        60
Secadora de ropa               58
Mesa de noche                  56
Bloques de construcción        56
Balón de baloncesto            55
Bicicleta                      54
Vaso térmico                   54
Refrigerador                   54
Impresora                      54
Batería                        53
Cubertería                     52
Carrito de control remoto      52
Modelado predictivo            51
Auriculares                    50
Ajedrez de madera              49
Iphone 15                      49
Lavavajillas                   48
Lavadora de ropa               47
Guitarra eléctrica             47
Guitarra acústica              46
Sofá reclinable                46
Silla de oficina               46
Estufa                         45
Cama box                       44
Kit de bancas                  44
Smart TV                       44
Cama king                      44
Cubo mágico 8x8                44
Mesa de centro                 44
Set de ollas                   43
Smartwatch                     43
Juego de mesa                  43
Balón de voleibol              42
Dashboards con Power BI        42
Tablet ABXY                    42
Sillón                         41
Iniciando en programación      41
Set de vasos                   41
Asistente virtual              40
Muñeca bebé                    40
Mesa de comedor                40
Dinosaurio Rex                 40
Cuerda para saltar             40
Mochila                        39
Ciencia de datos con Python    39
Pandereta                      36
Olla de presión                35
Auriculares con micrófono      33
Celular ABXY                   33
Name: count, dtype: int64

In [52]:
cantidad_productos_t2 = contador_categorias(tienda2.Producto)
cantidad_productos_t2

Producto
Iniciando en programación      65
Microondas                     62
Batería                        61
Guitarra acústica              58
Pandereta                      58
Secadora de ropa               57
Iphone 15                      55
Bloques de construcción        54
Armario                        54
Set de ollas                   52
Cuerda para saltar             50
Lavadora de ropa               49
Silla de oficina               49
Refrigerador                   49
Muñeca bebé                    48
Carrito de control remoto      47
Guitarra eléctrica             47
Cama box                       47
Vaso térmico                   47
Modelado predictivo            46
Balón de voleibol              46
Set de vasos                   46
Lavavajillas                   46
Kit de bancas                  45
Cama king                      45
Bicicleta                      45
Mochila                        45
Cubo mágico 8x8                45
Ajedrez de madera              45
Mesa de centro                 45
TV LED UHD 4K                  45
Tablet ABXY                    44
Smartwatch                     44
Mesa de noche                  44
Olla de presión                44
Auriculares con micrófono      44
Sofá reclinable                44
Dashboards con Power BI        43
Ciencia de datos con Python    43
Balón de baloncesto            42
Estufa                         42
Dinosaurio Rex                 42
Celular ABXY                   41
Smart TV                       40
Cubertería                     39
Asistente virtual              38
Auriculares                    37
Sillón                         35
Mesa de comedor                34
Impresora                      34
Juego de mesa                  32
Name: count, dtype: int64

In [23]:
cantidad_productos_t3 = contador_categorias(tienda3.Producto)
cantidad_productos_t3

Producto
Kit de bancas                  57
Mesa de comedor                56
Cama king                      56
Set de ollas                   55
Mesa de noche                  55
Smart TV                       54
Estufa                         53
Cuerda para saltar             53
Modelado predictivo            53
Carrito de control remoto      52
Lavavajillas                   52
Bicicleta                      52
Juego de mesa                  52
Pandereta                      51
Impresora                      50
Refrigerador                   50
Balón de baloncesto            50
Armario                        49
Balón de voleibol              48
Tablet ABXY                    48
Sillón                         47
Guitarra acústica              47
Olla de presión                47
Ajedrez de madera              47
Ciencia de datos con Python    46
Iphone 15                      46
Dinosaurio Rex                 46
Silla de oficina               46
Smartwatch                     46
Cama box                       46
Iniciando en programación      46
Secadora de ropa               45
Sofá reclinable                45
Celular ABXY                   44
Cubo mágico 8x8                44
Auriculares                    43
TV LED UHD 4K                  42
Mesa de centro                 42
Lavadora de ropa               42
Batería                        41
Dashboards con Power BI        40
Cubertería                     39
Muñeca bebé                    39
Auriculares con micrófono      39
Asistente virtual              39
Vaso térmico                   38
Guitarra eléctrica             38
Set de vasos                   36
Mochila                        36
Microondas                     36
Bloques de construcción        35
Name: count, dtype: int64

In [24]:
cantidad_productos_t4 = contador_categorias(tienda4.Producto)
cantidad_productos_t4

Producto
Cama box                       62
Cubertería                     59
Dashboards con Power BI        56
Cama king                      56
Carrito de control remoto      55
Mesa de comedor                55
Mesa de noche                  55
Smart TV                       54
Bloques de construcción        54
Pandereta                      52
Impresora                      50
Modelado predictivo            50
Asistente virtual              50
Set de ollas                   50
Balón de baloncesto            50
Vaso térmico                   48
Iphone 15                      48
Muñeca bebé                    48
Batería                        48
Balón de voleibol              48
Juego de mesa                  48
Microondas                     48
Estufa                         47
Cubo mágico 8x8                47
Sofá reclinable                47
Dinosaurio Rex                 47
Olla de presión                46
Set de vasos                   46
Tablet ABXY                    45
Sillón                         45
Bicicleta                      45
Mesa de centro                 44
Auriculares                    44
Lavavajillas                   43
Cuerda para saltar             43
Iniciando en programación      43
Mochila                        43
Auriculares con micrófono      42
Kit de bancas                  42
TV LED UHD 4K                  40
Secadora de ropa               40
Silla de oficina               40
Celular ABXY                   39
Ajedrez de madera              39
Smartwatch                     39
Lavadora de ropa               38
Refrigerador                   38
Ciencia de datos con Python    38
Guitarra acústica              37
Armario                        34
Guitarra eléctrica             33
Name: count, dtype: int64

In [63]:
identificador_tiendas = ['Tienda 1', 'Tienda 2', 'Tienda 3 ', 'Tienda 4']
total_productos = cantidad_productos_t1,cantidad_productos_t2,cantidad_productos_t3,cantidad_productos_t4
cantidad_total_productos = list(zip(identificador_tiendas, total_productos))
cantidad_total_productos

[('Tienda 1',
  Producto
  Microondas                     60
  TV LED UHD 4K                  60
  Armario                        60
  Secadora de ropa               58
  Mesa de noche                  56
  Bloques de construcción        56
  Balón de baloncesto            55
  Bicicleta                      54
  Vaso térmico                   54
  Refrigerador                   54
  Impresora                      54
  Batería                        53
  Cubertería                     52
  Carrito de control remoto      52
  Modelado predictivo            51
  Auriculares                    50
  Ajedrez de madera              49
  Iphone 15                      49
  Lavavajillas                   48
  Lavadora de ropa               47
  Guitarra eléctrica             47
  Guitarra acústica              46
  Sofá reclinable                46
  Silla de oficina               46
  Estufa                         45
  Cama box                       44
  Kit de bancas                  44
  S

# 5. Envío promedio por tienda

En este paso, el objetivo es calcular el costo de envío promedio para cada tienda.

In [27]:
costo_envio_t1 = tienda['Costo de envío']
costo_envio_t2 = tienda2['Costo de envío']
costo_envio_t3 = tienda3['Costo de envío']
costo_envio_t4 = tienda4['Costo de envío']

In [50]:
len(costo_envio_t1)


2359

In [51]:
promedio_envio_t1 = round(calcula_promedio(costo_envio_t1),1)
promedio_envio_t2 = round(calcula_promedio(costo_envio_t2),1)
promedio_envio_t3 = round(calcula_promedio(costo_envio_t3),1)
promedio_envio_t4 = round(calcula_promedio(costo_envio_t4),1)
print(promedio_envio_t1, promedio_envio_t2, promedio_envio_t3, promedio_envio_t4)

26018.6 25216.2 24805.7 23459.5


# Extra. Realiza un análisis de los datos para ayudar al Sr. Juan a tomar mejores decisiones.
### Crea un reporte con tus hallazgos para el cliente.

**Sr. Juan**, basados en los datos proporcionados y el análisis realizado podemos deducir que;

* Los ingresos estimados por la tienda 1 fueron: 1.150.880.400
* Los ingresos estimados por la tienda 2 fueron: 1.116.343.500
* Los ingresos estimados por la tienda 3 fueron: 1.098.019.600
* Los ingresos estimados por la tienda 4 fueron: 1.038.375.700

#### La tienda que generó mas ventas fue: **Tienda 1: 1.150.880.400**
#### La tienda que generó menos ventas fue: **Tienda 2: 1.038.375.700**

Tambien, podemos ver la categoria mas vendida en cada tienda;

### Tienda 1
*  Muebles                    465
*  Electrónicos               448
*  Juguetes                   324
*  Electrodomésticos          312
*  Deportes y diversión       284
*  Instrumentos musicales     182
*  Libros                     173
*  Artículos para el hogar    171

### Tienda 2
*  Muebles                    442
*  Electrónicos               422
*  Juguetes                   313
*  Electrodomésticos          305
*  Deportes y diversión       275
*  Instrumentos musicales     224
*  Libros                     197
*  Artículos para el hogar    181

### Tienda 3
*  Muebles                    499
*  Electrónicos               451
*  Juguetes                   315
*  Electrodomésticos          278
*  Deportes y diversión       277
*  Libros                     185
*  Instrumentos musicales     177
*  Artículos para el hogar    177

### Tienda 4
*  Muebles                    480
*  Electrónicos               451
*  Juguetes                   338
*  Deportes y diversión       277
*  Electrodomésticos          254
*  Artículos para el hogar    201
*  Libros                     187
*  Instrumentos musicales     170

En el siguiente análisis podemos verificar que tan satisfechos se encuentran los clientes:

- Tienda 1 - **3.98**/5
- Tienda 2 - **4.04**/5
- Tienda 3 - **4.05**/5
- Tienda 4 - **4.0**/5


Como penultimo dato, podemos ver un listado de los productos mas y menos vendidos por tienda:

In [64]:
cantidad_total_productos

[('Tienda 1',
  Producto
  Microondas                     60
  TV LED UHD 4K                  60
  Armario                        60
  Secadora de ropa               58
  Mesa de noche                  56
  Bloques de construcción        56
  Balón de baloncesto            55
  Bicicleta                      54
  Vaso térmico                   54
  Refrigerador                   54
  Impresora                      54
  Batería                        53
  Cubertería                     52
  Carrito de control remoto      52
  Modelado predictivo            51
  Auriculares                    50
  Ajedrez de madera              49
  Iphone 15                      49
  Lavavajillas                   48
  Lavadora de ropa               47
  Guitarra eléctrica             47
  Guitarra acústica              46
  Sofá reclinable                46
  Silla de oficina               46
  Estufa                         45
  Cama box                       44
  Kit de bancas                  44
  S

Finalmente, tenemos el promedio de envío por cada tienda:

- Tienda 1 - 26018.6
- Tienda 2 - 25216.2
- Tienda 3 - 24805.7
- Tienda 4 - 23459.5